gnd/src/main/java/com/google/android/gnd/service/firestore/FirestoreDataStore.java

```diff
@@ -41,21 +42,23 @@
 import javax.inject.Singleton;
 
 @Singleton
-public class FirestoreDataStore implements RemoteDataStore {
+public class FirestoreDataStore implements RemoteDataStore, OfflineUuidGenerator {
```

Comment:
Instead of implementing the interface in `FirestoreDataStore` itself, how about writing a separate `FirestoreUuidGenerator` class (that also depends on the Firestore instance)? That would separate functionality and make the object dependency graph cleaner for the classes that only need to generate IDs (e.g. where a local entity is created and to be stored locally only) and don't otherwise interact with Firestore (and so don't need to pull in the whole dependency structure of the data store).

Reply:
Ideally yes. To that we'd also need to refactor the Firestore api client instance setup being done in the constructor. Wanted to keep these PRs small, so decided to postpone that for now. Ok for you? Note that this is all part of the implementation details of our Firestore impl. If someone were to implement their own backend data store, they'd be free to split the implementation into two classes or provide a single impl that implements both interfaces.

Reply:
Sure. Maybe add a TODO to split this up.

Reply:
Done!